src/run_model.jl

```diff
@@ -29,11 +29,14 @@ function run_model(::Type{T},P::Parameter) where {T<:AbstractFloat}
     G = Grid{T,Tprog}(P)
     C = Constants{T,Tprog}(P,G)
     F = Forcing{T}(P,G)
-    S = ModelSetup{T,Tprog}(P,G,C,F)
 
-    Prog = initial_conditions(Tprog,S)
+    Prog = initial_conditions(Tprog,G,P,C)
     Diag = preallocate(T,Tprog,G)
 
-    Prog = time_integration(Prog,Diag,S)
+    # one structure with everything already inside 
+    S = ModelSetup{T,Tprog}(P,G,C,F,Prog,Diag,0)
```

Comment:
It seems that this line is what's making one test fail, specifically line 37 of `runtests.jl` doing `@testset "Mixed Precision"`. I get the error 
``` julia
 MethodError: Cannot `convert` an object of type 
    ShallowWaters.PrognosticVars{Float64} to an object of type 
    ShallowWaters.PrognosticVars{Float32}
```
All other tests in `runtest.jl` pass. This one really confuses me though: I did indeed define the prognostic variables to be of type Float64, so it makes sense that I can't convert to a Float32 (I think?) but when I look at the original code on the main branch it seems like everything is defined in the same way. I would have expected that I flipped a `T` and a `Tprog` somewhere, but I can't find it. So I'm confused about why this test fails here in my branch. (I.e., during `run_model`, `initial_conditions` is passed `Tprog`, and in the test this is `Float64`, so it should still be the case that `Prog` is defined as `ShallowWaters.PrognosticVars{Float64}` in the main branch, which I would think should also fail). I can keep looking later but I'm a bit stumped right now